client/my-sites/checkout/composite-checkout/components/checkout-main.tsx

```diff
@@ -126,14 +125,10 @@ export default function CheckoutMain( {
 } ) {
 	const translate = useTranslate();
 	const isJetpackNotAtomic =
-		useSelector(
-			( state ) => siteId && isJetpackSite( state, siteId ) && ! isAtomicSite( state, siteId )
-		) ||
-		isJetpackCheckout ||
-		false;
-	const hasJetpackStandalonePlugins =
-		useSelector( ( state ) => siteId && isJetpackProductSite( state, siteId ) ) || false;
-	const usesJetpackProducts = isJetpackNotAtomic || hasJetpackStandalonePlugins;
+		useSelector( ( state ) => {
+			// isJetpackSite already checks for isAtomicSite
+			return siteId && isJetpackSite( state, siteId );
+		} ) || isJetpackCheckout;
```

Comment:
For the payments team: `isJetpackSite` now does what `isJetpackProductSite` did before and it also checks for `isAtomicSite`, thus that check was also removed from here.